modules/c++/nitf/source/ImageWriter.cpp

```diff
@@ -26,27 +26,24 @@ using namespace nitf;
 
 ImageWriter::ImageWriter(nitf::ImageSubheader& subheader)
 {
-    setNative(nitf_ImageWriter_construct(subheader.getNative(), NULL, &error));
+    auto* writer =
+            nitf_ImageWriter_construct(subheader.getNative(), NULL, &error);
+    setNativeOrThrow(writer, &error);
```

Comment:
It's not that we're getting a null native handle but it's failing in the setup.   The error member needs to be checked when nitf_ImageWriter_construct returns

```suggestion
    setNativeOrThrow(writer, &error);
    if( error.level != 0 )
       throw nitf::NITFException(error);
```

Reply:
It should be the case that `nitf_ImageWriter_construct` returns NULL if the error is set. I've updated that function to that effect.